packages/api-format/src/codec/base/Array.ts

```diff
@@ -0,0 +1,111 @@
+// Copyright 2017-2018 @polkadot/api-format authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { Base } from '../types';
+
+import u8aConcat from '@polkadot/util/u8a/concat';
+
+// NOTE or LengthCompact
+import Length from './Length';
+
+export default class CodecArray <T extends Base<any>> implements Base<Array<T>> {
+  private _length: Length;
+  private _Type: { new(value?: any): T };
+
+  raw: Array<T>;
+
+  constructor (Type: { new(value?: any): T }, value: Array<any> = [] as Array<any>) {
+    this._length = new Length(value.length);
+    this._Type = Type;
+    this.raw = value.map((entry) =>
+      new this._Type(entry)
+    );
+  }
+
+  static with <O extends Base<any>> (Type: { new(value?: any): O }): { new(value?: any): CodecArray<O> } {
```

Comment:
Shortcut function so we can have less noise when actually constructing these inline